File: Java/1346. Check If N and Its Double Exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> set = new HashSet<>();
        int zero = 0;
        
        for(int num: arr){
            if(num == 0){
                zero++;
            }
            set.add(num);
        }
        
        for(int num: set){
            if(num == 0){
                if(zero > 1){
                    return true;
                }
            }else if(set.contains(num*2)){
                //System.out.println(num);
                return true;
            }
        }
        
        
        return false;
    }
}

File: Python3/1346. Check If N and Its Double Exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        
        # if num == 0, check if there are more than one 0
        # otherwise, search the set for num*2
        sets = set(arr)
        
        for num in arr:
            if num == 0:
                if arr.count(0) > 1:
                    return True
            
            elif num*2 in sets:
                #print(num, num*2)
                return True
            
        return False
            
            
                